### DISCUSSION TITLE
Messages are sent too fast

### DISCUSSION THREAD

**Discussion by rahav** [contiki-ng/contiki-ng#2325] [Q&A] [answered]
Hello and thank you in advance for any thought you put into my question.

I noticed that messages are dropped when sent close to each other in time. Turning on debug messages for RPL and MAC I see the following:

    20:20.929	ID:3	[DBG : Device    ] Sender = 3 (Unicast to fd00::202:2:2:2) msg = 21, at = 1220001; [DED] Length = 28
    20:20.929	ID:3	[INFO: RPL       ] creating hop-by-hop option
    20:20.929	ID:3	[DBG : RPL       ] no SRH found
    ...
    20:20.968	ID:1	[INFO: CSMA      ] received packet from 0003.0003.0003.0003, seqno 29, len 54
    20:20.968	ID:1	[INFO: RPL       ] ext hdr: packet from fd00::203:3:3:3 to fd00::202:2:2:2 going up, sender closer 0 (256 < 128), rank error 0, loop detected 0
    20:20.968	ID:1	[INFO: RPL       ] SRH creating source routing header with destination fd00::202:2:2:2 
    20:20.968	ID:1	[INFO: RPL       ] SRH path len: 0, ComprI 15, ComprE 15, ext len 8 (padding 0)
    20:20.968	ID:1	[WARN: CSMA      ] could not allocate neighbor, dropping packet
    20:20.968	ID:1	[INFO: RPL       ] packet sent to 0002.0002.0002.0002, status 6, tx 1, new link metric 128

To mitigate I added scheduled transmissions and I saw improvement as i increased the timer durations. However, I'm already at `200 ms` between messages, which is better but still exhibit this behavior. I'm using RPL Lite, UDP, and unicast.

1. Is there a rule of thumb for how part apart message sending should be?
2. Is there another way to solve the problem?

**Comment by arurke** [comment, marked as answer]
I am not familiar with the CSMA lib., but glancing at `csma-output.c` it looks like _"could not allocate neighbor, dropping packet"_ indicates there is no room to create new neighbor entries. The default max. number of simultaneous neighbors (with non-empty queues) is 2. Try to increase `CSMA_CONF_MAX_NEIGHBOR_QUEUES` to match you use-case and see if that helps?

**Reply by rahav** [reply]
Thank YOU so much. This was it, I increased it to 4 and saw a dramatic improvement.
